legate/core/launcher.py

```diff
@@ -538,64 +670,77 @@ def update_storages(self):
-        mapper_id=0,
-        tag=0,
-        error_on_interference=True,
-        side_effect=False,
-    ):
+        context: Context,
+        task_id: int,
+        mapper_id: int = 0,
+        tag: int = 0,
+        error_on_interference: bool = True,
+        side_effect: bool = False,
+    ) -> None:
         assert type(tag) != bool
         self._context = context
         self._runtime = context.runtime
         self._core_types = self._runtime.core_context.type_system
         self._task_id = task_id
         self._mapper_id = mapper_id
-        self._inputs = []
-        self._outputs = []
-        self._reductions = []
-        self._scalars = []
-        self._comms = []
+        self._inputs: list[Union[RegionFieldArg, FutureStoreArg]] = []
+        self._outputs: list[Union[RegionFieldArg, FutureStoreArg]] = []
+        self._reductions: list[Union[RegionFieldArg, FutureStoreArg]] = []
+        self._scalars: list[ScalarArg] = []
+        self._comms: list[FutureMap] = []
         self._req_analyzer = RequirementAnalyzer(error_on_interference)
         self._out_analyzer = OutputAnalyzer(context.runtime)
-        self._future_args = list()
-        self._future_map_args = list()
+        self._future_args: list[Future] = []
+        self._future_map_args: list[FutureMap] = []
         self._tag = tag
-        self._sharding_space = None
-        self._point = None
-        self._output_regions = list()
+        self._sharding_space: Union[IndexSpace, None] = None
+        self._point: Union[Point, None] = None
+        self._output_regions: list[OutputRegion] = []
         self._error_on_interference = error_on_interference
         self._has_side_effect = side_effect
 
     @property
-    def library_task_id(self):
+    def library_task_id(self) -> int:
         return self._task_id
 
     @property
-    def library_mapper_id(self):
+    def library_mapper_id(self) -> int:
         return self._mapper_id
 
     @property
-    def legion_task_id(self):
+    def legion_task_id(self) -> int:
         return self._context.get_task_id(self._task_id)
 
     @property
-    def legion_mapper_id(self):
+    def legion_mapper_id(self) -> int:
         return self._context.get_mapper_id(self._mapper_id)
 
-    def __del__(self):
+    def __del__(self) -> None:
         del self._req_analyzer
         del self._out_analyzer
         self._future_args.clear()
         self._future_map_args.clear()
         self._output_regions.clear()
 
-    def add_scalar_arg(self, value, dtype, untyped=True):
+    def add_scalar_arg(
+        self,
+        value: Any,
+        dtype: Union[bool, pa.lib.DataType],
+        untyped: bool = True,
+    ) -> None:
         self._scalars.append(
             ScalarArg(self._core_types, value, dtype, untyped)
         )
 
-    def add_store(self, args, store, proj, perm, tag, flags):
+    def add_store(
+        self,
+        args: list[Union[RegionFieldArg, FutureStoreArg]],  # TODO: (bev) ABC?
```

Comment:
And it looks like you have the same idea. :)

Reply:
https://github.com/nv-legate/legate.core/issues/192